cms/models.py

```diff
@@ -12,6 +13,65 @@
 from wagtail.images.blocks import ImageChooserBlock
 
 
+class CourseProgramChildPage(Page):
+    """
+    Abstract page representing a child of Course/Program Page
+    """
+
+    class Meta:
+        abstract = True
+
+    parent_page_types = ["CoursePage", "ProgramPage"]
+
+    @classmethod
+    def can_create_at(cls, parent):
+        # You can only create one of these page under course / program.
+        return (
+            super(CourseProgramChildPage, cls).can_create_at(parent)
+            and parent.get_children().type(cls).count() == 0
+        )
+
+
+class WhoShouldEnrollPage(CourseProgramChildPage):
+    """
+    Who should enroll child page for "Who Should Enroll" section.
+    """
+
+    subpage_types = []
+
+    image = models.ForeignKey(
+        Image,
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name="+",
+        help_text="Image size must be at least 870x500 pixels.",
+    )
+    content = StreamField(
+        [
+            (
+                "item",
+                blocks.RichTextBlock(
+                    icon="plus", features=["bold", "italic", "ol", "ul"]
+                ),
+            )
+        ],
+        blank=False,
+        help_text='Contents of the "Who Should Enroll" setion.',
+    )
+
+    content_panels = [StreamFieldPanel("content"), FieldPanel("image")]
+
+    # disable promote panels, no need for slug entry, it will be autogenerated
+    promote_panels = []
+
+    def save(self, *args, **kwargs):
+        # autogenerate a unique slug so we don't hit a ValidationError
+        self.title = "Who Should Enroll"
+        self.slug = slugify("{}-{}".format(self.get_parent().id, self.title))
+        super().save()
```

Comment:
did you check it ? It is throwing a validation error on my end. I think we should keep the `title` that would be the title of page.

Reply:
Do you mean this actual implementation, when checked out on your system, is throwing a `ValidationError`?

Reply:
do not forget to pass  *args, **kwargs to parent. 
`super(WhoShouldEnrollPage, self).save(*args, **kwargs)`

Reply:
Done.